### PR TITLE
These changes are correct?

### PR DESCRIPTION
**Pull request by ranbir7**
I didnt change anything further 
since I needed ur approval if everything is 
going well...

### CODE REVIEW

**Comment by ranbir7** [comment]
If changes look good 
merge it, I'll add further changes !😁

File: autoprof/image/image_object.py [hunk 1]
```diff
@@ -25,8 +26,42 @@ class BaseImage(object):
         origin
     """
 
-    def __init__(self, data = None, pixelscale = None, window = None, filename = None, zeropoint = None, note = None, origin = None, center = None, **kwargs):
-        
+    def __init__(self, 
+            data: Optional[torch.Tensor] = None,
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: data can be given as `torch.Tensor` or as `np.ndarray`

File: autoprof/image/image_object.py [hunk 2]
```diff
@@ -25,8 +26,42 @@ class BaseImage(object):
...
-    def __init__(self, data = None, pixelscale = None, window = None, filename = None, zeropoint = None, note = None, origin = None, center = None, **kwargs):
-        
+    def __init__(self, 
+            data: Optional[torch.Tensor] = None, 
+            pixelscale: Optional[Union[float, torch.Tensor]] = None,
+            window: Optional[Window] = None, 
+            filename: Optional[str] = None, 
+            zeropoint: Optional[Union[float, torch.Tensor]] = None, 
+            note: Optional[str] = None, 
+            origin: Optional[torch.Tensor] = None,
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: `torch.Tensor` or `np.ndarray`

**Comment by ranbir7** [comment]
Ahh! here we go again, CodeCoverage fails...

File: autoprof/image/image_object.py [hunk 3]
```diff
@@ -25,8 +26,42 @@ class BaseImage(object):
...
-        
+    def __init__(self, 
+            data: Optional[torch.Tensor] = None, 
+            pixelscale: Optional[Union[float, torch.Tensor]] = None,
+            window: Optional[Window] = None, 
+            filename: Optional[str] = None, 
+            zeropoint: Optional[Union[float, torch.Tensor]] = None, 
+            note: Optional[str] = None, 
+            origin: Optional[torch.Tensor] = None, 
+            center: Optional[torch.Tensor] = None,
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: `torch.Tensor` or `np.ndarray`

File: autoprof/image/image_object.py [hunk 4]
```diff
@@ -25,8 +26,42 @@ class BaseImage(object):
...
+        window : Window or None, optional
+            A Window object defining the area of the image to use. Default is None.
+        filename : str or None, optional
+            The name of a file containing the image data. Default is None.
+        zeropoint : float or None, optional
+            The image's zeropoint, used for flux calibration. Default is None.
+        note : str or None, optional
+            A note describing the image. Default is None.
+        origin : numpy.ndarray or None, optional
+            The origin of the image in the coordinate system, as a 2D array of length 2. Default is None.
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: the origin is a `1D` array or tensor of length 2

File: autoprof/image/image_object.py [hunk 5]
```diff
@@ -25,8 +26,42 @@ class BaseImage(object):
...
+        center : numpy.ndarray or None, optional
+            The center of the image in the coordinate system, as a 2D array of length 2. Default is None.
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: The center is a `1D` array or tensor of length 2

File: autoprof/image/image_object.py [hunk 6]
```diff
@@ -55,12 +90,30 @@ def __init__(self, data = None, pixelscale = None, window = None, filename = Non
             
     @property
     def origin(self):
+        """
+        Returns the origin (bottom-left corner) of the image window.
+
+        Returns:
+            torch.Tensor: A 2D tensor of shape (2,) containing the (x, y) coordinates of the origin.
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: origin is a `1D` tensor

File: autoprof/image/image_object.py [hunk 7]
```diff
@@ -55,12 +90,30 @@ def __init__(self, data = None, pixelscale = None, window = None, filename = Non
...
+        """
         return self.window.origin
     @property
     def shape(self):
+        """
+        Returns the shape (size) of the image window.
+            
+        Returns:
+                torch.Tensor: A 2D tensor of shape (2,) containing the (width, height) of the window in pixels.
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: shape is a `1D` tensor

File: autoprof/image/image_object.py [hunk 8]
```diff
@@ -55,12 +90,30 @@ def __init__(self, data = None, pixelscale = None, window = None, filename = Non
...
+        """
         return self.window.shape
     @property
     def center(self):
+        """
+        Returns the center of the image window.
+
+        Returns:
+            torch.Tensor: A 2D tensor of shape (2,) containing the (x, y) coordinates of the center.
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: center is a `1D` tensor

**Review by ConnorStoneAstro — COMMENTED** [review]
This looks great! Just some very minor points to address

**Comment by ConnorStoneAstro** [comment]
Also, you can ignore the code coverage issues, for pull requests you only need to look at the unit tests, code coverage is for me

File: autoprof/image/image_object.py (hunk 2 above)
Comment (ranbir7) [reply autoprof/image/image_object.py]: origin or zero point? which arg?

Comment (ConnorStoneAstro) [reply autoprof/image/image_object.py]: origin needs tensor and array. zeropoint is correct!

Comment (ranbir7) [reply autoprof/image/image_object.py]: can you please send me the link of typing module which u sent me earlier?

Comment (ConnorStoneAstro) [reply autoprof/image/image_object.py]: Was it this one: https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html

in this case I think you would just do `Optional[Union[np.ndarray, torch.Tensor]]`

Comment (ranbir7) [reply autoprof/image/image_object.py]: I did the exact thing I guess....yes this was the link !
Thanks!

**Comment by ranbir7** [comment]
Perfect ! 
Ill do further changes after u merge it....

**Comment by ConnorStoneAstro** [comment]
Looking good! Just the `center` type hint is left

**Comment by ranbir7** [comment]
wait a min....

**Comment by ranbir7** [comment]
Sheesh I made quite a few mistakes !
solving it right now..

**Comment by ranbir7** [comment]
The center should be an array?
since it could be  a list of two floats or tuple of two floats
I think it should be this right?
_Optional[Union[Tuple[float, float], List[float, float]]]_

**Comment by ConnorStoneAstro** [comment]
Hmmm, you are right. perhaps center and origin should be any iterable. In which case they would be `Sequence` type

**Comment by ranbir7** [comment]
I'm little confused....
what shall I write then?
it can also take torch.Tensor too
shall I make it 
_Optional[Union[Tuple[float, float], List[float, float], torch.Tensor]]_

**Comment by ConnorStoneAstro** [comment]
I think the type hint for both `origin` and `center` can be just `Optional[Sequence]`

**Comment by ranbir7** [comment]
So I'll just leave it like this?
_center : Optional[Sequence]_

**Comment by ConnorStoneAstro** [comment]
yes. To be more complete, it would be: `center: Optional[Sequence] = None` and the same for `origin`

**Comment by ConnorStoneAstro** [comment]
Oh, I think I was unclear, sorry it would be:
`origin: Optional[Sequence] = None`
`center: Optional[Sequence] = None`

**Comment by ranbir7** [comment]
It's Okay 
I got it 
I was In a hurry to sleep 
I messed it up !!

**Review by ConnorStoneAstro — APPROVED** [review]
This looks great! I'll merge it soon

**Pull request merged**